### ISSUE TITLE
Blockchain push panic during devnet execution 

### ISSUE THREAD

**Issue by jsdanielh** [nimiq/core-rs-albatross#957]
In the latest devnet execution a panic was observed in the `blockchain` `push` function:
```
Jul 25 09:22:33 validator1 nimiq-client[99778]: 2022-07-25T09:22:33.751971Z ERROR panic                | thread 'tokio-runtime-worker' panicked at 'Trying to rebranch across macro block': blockchain/src/blockchain/push.rs:405
Jul 25 09:22:33 validator1 nimiq-client[99778]:    0: <backtrace::capture::Backtrace as core::default::Default>::default
Jul 25 09:22:33 validator1 nimiq-client[99778]:    1: log_panics::Config::install_panic_hook::{{closure}}
Jul 25 09:22:33 validator1 nimiq-client[99778]:    2: std::panicking::rust_panic_with_hook
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/panicking.rs:702:17
Jul 25 09:22:33 validator1 nimiq-client[99778]:    3: std::panicking::begin_panic_handler::{{closure}}
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/panicking.rs:586:13
Jul 25 09:22:33 validator1 nimiq-client[99778]:    4: std::sys_common::backtrace::__rust_end_short_backtrace
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/sys_common/backtrace.rs:138:18
Jul 25 09:22:33 validator1 nimiq-client[99778]:    5: rust_begin_unwind
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/panicking.rs:584:5
Jul 25 09:22:33 validator1 nimiq-client[99778]:    6: core::panicking::panic_fmt
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/core/src/panicking.rs:142:14
Jul 25 09:22:33 validator1 nimiq-client[99778]:    7: nimiq_blockchain::blockchain::push::<impl nimiq_blockchain::blockchain::blockchain::Blockchain>::rebranch
Jul 25 09:22:33 validator1 nimiq-client[99778]:    8: nimiq_blockchain::blockchain::push::<impl nimiq_blockchain::blockchain::blockchain::Blockchain>::do_push
Jul 25 09:22:33 validator1 nimiq-client[99778]:    9: nimiq_blockchain::blockchain::push::<impl nimiq_blockchain::blockchain::blockchain::Blockchain>::push_wrapperfn
Jul 25 09:22:33 validator1 nimiq-client[99778]:   10: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
Jul 25 09:22:33 validator1 nimiq-client[99778]:   11: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
Jul 25 09:22:33 validator1 nimiq-client[99778]:   12: tokio::runtime::task::harness::Harness<T,S>::poll
Jul 25 09:22:33 validator1 nimiq-client[99778]:   13: tokio::runtime::blocking::pool::Inner::run
Jul 25 09:22:33 validator1 nimiq-client[99778]:   14: std::sys_common::backtrace::__rust_begin_short_backtrace
Jul 25 09:22:33 validator1 nimiq-client[99778]:   15: core::ops::function::FnOnce::call_once{{vtable.shim}}
Jul 25 09:22:33 validator1 nimiq-client[99778]:   16: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/alloc/src/boxed.rs:1951:9
Jul 25 09:22:33 validator1 nimiq-client[99778]:       <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/alloc/src/boxed.rs:1951:9
Jul 25 09:22:33 validator1 nimiq-client[99778]:       std::sys::unix::thread::Thread::new::thread_start
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/sys/unix/thread.rs:108:17
Jul 25 09:22:33 validator1 nimiq-client[99778]:   17: start_thread
Jul 25 09:22:33 validator1 nimiq-client[99778]:   18: clone
```
Then it causes the following panic:
```
ul 25 09:22:33 validator1 nimiq-client[99778]: 2022-07-25T09:22:33.989970Z ERROR panic                | thread 'tokio-runtime-worker' panicked at 'blockchain.push() should not panic: JoinError::Panic(Id(7432049), ...)': consensus/src/sync/block_queue.rs:368
Jul 25 09:22:33 validator1 nimiq-client[99778]:    0: <backtrace::capture::Backtrace as core::default::Default>::default
Jul 25 09:22:33 validator1 nimiq-client[99778]:    1: log_panics::Config::install_panic_hook::{{closure}}
Jul 25 09:22:33 validator1 nimiq-client[99778]:    2: std::panicking::rust_panic_with_hook
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/panicking.rs:702:17
Jul 25 09:22:33 validator1 nimiq-client[99778]:    3: std::panicking::begin_panic_handler::{{closure}}
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/panicking.rs:588:13
Jul 25 09:22:33 validator1 nimiq-client[99778]:    4: std::sys_common::backtrace::__rust_end_short_backtrace
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/sys_common/backtrace.rs:138:18
Jul 25 09:22:33 validator1 nimiq-client[99778]:    5: rust_begin_unwind
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/panicking.rs:584:5
Jul 25 09:22:33 validator1 nimiq-client[99778]:    6: core::panicking::panic_fmt
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/core/src/panicking.rs:142:14
Jul 25 09:22:33 validator1 nimiq-client[99778]:    7: core::result::unwrap_failed
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/core/src/result.rs:1805:5
Jul 25 09:22:33 validator1 nimiq-client[99778]:    8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
Jul 25 09:22:33 validator1 nimiq-client[99778]:    9: futures_util::stream::stream::StreamExt::poll_next_unpin
Jul 25 09:22:33 validator1 nimiq-client[99778]:   10: <nimiq_consensus::consensus::Consensus<N> as core::future::future::Future>::poll
Jul 25 09:22:33 validator1 nimiq-client[99778]:   11: <tokio_metrics::task::Instrumented<T> as core::future::future::Future>::poll
Jul 25 09:22:33 validator1 nimiq-client[99778]:   12: tokio::runtime::task::harness::Harness<T,S>::poll
Jul 25 09:22:33 validator1 nimiq-client[99778]:   13: std::thread::local::LocalKey<T>::with
Jul 25 09:22:33 validator1 nimiq-client[99778]:   14: tokio::runtime::thread_pool::worker::Context::run_task
Jul 25 09:22:33 validator1 nimiq-client[99778]:   15: tokio::runtime::thread_pool::worker::Context::run
Jul 25 09:22:33 validator1 nimiq-client[99778]:   16: tokio::macros::scoped_tls::ScopedKey<T>::set
Jul 25 09:22:33 validator1 nimiq-client[99778]:   17: tokio::runtime::thread_pool::worker::run
Jul 25 09:22:33 validator1 nimiq-client[99778]:   18: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
Jul 25 09:22:33 validator1 nimiq-client[99778]:   19: tokio::runtime::task::harness::Harness<T,S>::poll
Jul 25 09:22:33 validator1 nimiq-client[99778]:   20: tokio::runtime::blocking::pool::Inner::run
Jul 25 09:22:33 validator1 nimiq-client[99778]:   21: std::sys_common::backtrace::__rust_begin_short_backtrace
Jul 25 09:22:33 validator1 nimiq-client[99778]:   22: core::ops::function::FnOnce::call_once{{vtable.shim}}
Jul 25 09:22:33 validator1 nimiq-client[99778]:   23: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/alloc/src/boxed.rs:1951:9
Jul 25 09:22:33 validator1 nimiq-client[99778]:       <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/alloc/src/boxed.rs:1951:9
Jul 25 09:22:33 validator1 nimiq-client[99778]:       std::sys::unix::thread::Thread::new::thread_start
Jul 25 09:22:33 validator1 nimiq-client[99778]:              at /rustc/43347397f7c5ca9a670a3bb3890c7187e24a52ab/library/std/src/sys/unix/thread.rs:108:17
Jul 25 09:22:33 validator1 nimiq-client[99778]:   24: start_thread
Jul 25 09:22:33 validator1 nimiq-client[99778]:   25: clone
```

**Comment by viquezclaudio** [comment]
Moving to a different history sync implementation

**Comment by jsdanielh** [comment]
Fixed with the revert done in e37f3ac312d1936048a9fc33427919f74c4ff63e.
